### ISSUE TITLE
Back-Translation loss + Best model selection

### ISSUE THREAD

**Issue by bakszero** [facebookresearch/XLM#125]
Hi, I'm facing the following problem when I have a custom language for which no parallel data is available (only scarce monolingual  data).
I'm attempting to do Unsupervised MT from English to the scarce language. I have a few questions:

1. For the fine-tuning stage for unsupervised MT(after the model has been pretrained), I find that I'm unable to use the backtranslation loss (`--bt_steps`) as I get an error for need of parallel data for this step. I'm able to use the autoencoding loss though. Is this an inherent nature of backtranslation that parallel data be present for this step? Any pointers to a paper on this would be great.

2. Picking up from here, https://github.com/facebookresearch/XLM/issues/94, the best model selection during finetuning is done via selecting the validation BLEU on a parallel data set. Since my setup is completely unsupervised, what would be my `--stopping_criterion` in this case? Or should I just set `--max_epoch` manually?

3. Adding on to (2), how would I set my own custom loss metric by which I can select the best model according to the custom defined loss metric?

Btw, just noticed this repo has >100 issues which you guys have resolved. Thanks for the effort, really appreciate it!

**Comment by glample** [comment]
Indeed, the back-translation validation metric is not implemented there. In the current state, the code cannot select models without at least a small validation set, sorry about that.

One solution could be to re-implement it in `evaluator.py` (it should not be too hard by adding in `evaluator.py` something similar to the `bt_step` function in `trainer.py`). The new function would save something like `valid_en_fr_en_bleu` in the score metrics, then the stopping criterion can be set to: `--stopping_criterion valid_en_fr_en_bleu,10`
